plugins/module_utils/openssh/certificate.py

```diff
@@ -326,8 +336,17 @@ def cert_type(self, cert_type):
     def signing_key_fingerprint(self):
         return fingerprint(self.signing_key)
 
-    @abc.abstractmethod
     def public_key_fingerprint(self):
+        if self.public_key_type_string is None:
+            return b''
+        writer = _OpensshWriter()
+
+        writer.string(self.public_key_type_string)
+        self.write_public_key_params(writer)
+        return fingerprint(writer.bytes())
+
+    @abc.abstractmethod
+    def write_public_key_params(self, writer):
```

Comment:
This is a breaking API change. You need to remove `@abc.abstractmethod` and add in a docstring that in community.crypto 3.0.0 this will become an abstract method.

Also you should add a docstring here explaining what this function is expected to do.

Reply:
@jnohlgard I think it would be better to just introduce the feature in this PR and move the refactor to a follow up if necessary. As @felixfontein is indicating you would force this feature to wait until the next major release if it includes breaking changes.